GTG/gtk/preferences.py

```diff
@@ -75,74 +72,64 @@ def disable_gtg_autostart():
-        self.hbox1 = builder.get_object("hbox1")
+        builder.add_from_file(self.PREFERENCES_UI_FILE)
+
+        self.window = builder.get_object("Preferences")
+        self.autostart_button = builder.get_object("autostart_button")
+        self.preview_button = builder.get_object("preview_button")
+        self.bg_color_button = builder.get_object("bg_color_button")
         self.shortcut_button = builder.get_object("shortcut_button")
+        self.font_button = builder.get_object("font_button")
+        self.shortcut_popover = builder.get_object("shortcut_popover")
+        self.set_shortcut = builder.get_object("set_shortcut")
 
         self.shortcut = ShortcutWidget(builder)
-        help.add_help_shortcut(self.dialog, "preferences")
 
-        self.fontbutton = builder.get_object("fontbutton")
         self.timer = vmanager.timer
-        self.color_invalid = Color(50000, 0, 0)
-        self.refresh_hour = builder.get_object("hour")
-        self.refresh_mins = builder.get_object("min")
+        self.refresh_time = builder.get_object("time_entry")
         editor_font = self.config.get("font_name")
         if editor_font == "":
-            font = self.dialog.get_style_context().get_font(
+            font = self.window.get_style_context().get_font(
                 Gtk.StateFlags.NORMAL)
             editor_font = font.to_string()
-        self.fontbutton.set_font_name(editor_font)
-
-        builder.connect_signals({
-                                'on_pref_autostart_toggled':
-                                self.on_autostart_toggled,
-                                'on_pref_show_preview_toggled':
-                                self.toggle_preview,
-                                'on_bg_color_toggled':
-                                self.on_bg_color_toggled,
-                                'on_prefs_help':
-                                self.on_help,
-                                'on_prefs_close':
-                                self.on_close,
-                                'on_PreferencesDialog_delete_event':
-                                self.on_close,
-                                'on_fontbutton_font_set':
-                                self.on_font_change,
-                                'on_shortcut_button_toggled':
-                                self.shortcut.on_shortcut_toggled,
-                                'on_valid_check':
-                                self.valid_check,
-                                })
+        self.font_button.set_font_name(editor_font)
+
+        builder.connect_signals(self)
+
+        self.headerbar = builder.get_object("right_header_bar")
+        self.general = builder.get_object("general_pref_window")
+        self.plugins = builder.get_object("plugins")
+        self.synchronisation = builder.get_object("synchronisation")
+        self.stack = builder.get_object("stack")
+
+    def on_shortcut_popover(self, widget):
+        self.shortcut_popover.show()
```

Comment:
The same thing as was in editor toolbar. We could apply that logic, if I remember correctly, only to MenuButtons, but not normal Buttons and we triggered it like this, from python. Moreover, I guess this function will get more functionality about setting shortcut -- soon -- so I think it's fine